packages/ssx-react/src/next/backend.ts

```diff
@@ -0,0 +1,91 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getCsrfToken } from 'next-auth/react';
+import { SSXServer, SSXServerConfig } from '@spruceid/ssx-server';
+
+export const SSXNextAuth = (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  ssxConfig: SSXServerConfig = {}
+) => {
+  const credentials = {
+    message: {
+      label: 'Message',
+      type: 'text',
+      placeholder: '0x0',
+    },
+    signature: {
+      label: 'Signature',
+      type: 'text',
+      placeholder: '0x0',
+    },
+    daoLogin: {
+      label: 'daoLogin',
+      type: 'boolean',
+      placeholder: 'false',
+    },
+    resolveEns: {
+      label: 'resolveEns',
+      type: 'boolean',
+      placeholder: 'false',
+    },
+  };
+
+  const authorize = async credentials => {
+    try {
+      console.log('authorize');
+      console.log(credentials);
+
+      const ssx = new SSXServer(ssxConfig);
+      const nonce = await getCsrfToken({ req });
+
+      // validate signature, nonce and ssx config options
+      const { success, error, session } = await ssx.login(
+        credentials?.message || '',
+        credentials?.signature || '',
+        JSON.parse(credentials?.daoLogin || 'false'),
+        JSON.parse(credentials?.resolveEns || 'false'),
+        nonce || ''
+      );
+      const { siwe, signature, daoLogin, ens } = session;
+      if (!siwe) return null;
+
+      //  // check domain
```

Comment:
Do we want to include and enforce same domain validation? This looks like a next pattern. If we want to do this, we need to get the domain from the dev, perhaps as an ENV var

Reply:
We will not check the domain. The nonce check should be sufficient